### PR TITLE
Support generate net-att-def for ib-sriov CNI

### PR DESCRIPTION
**Pull request by pliurh**

### CODE REVIEW

File: deploy/crds/sriovnetwork.openshift.io_v1_sriovibnetwork_cr.yaml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+apiVersion: sriovnetwork.openshift.io/v1
+kind: SriovIBNetwork
+metadata:
+  name: example-sriovibnetwork
```
[the remark this answers is not in the document]
Comment (pliurh) [reply deploy/crds/sriovnetwork.openshift.io_v1_sriovibnetwork_cr.yaml:4]: We use the namespace `openshift-sriov-network-operator` for openshift, whereas `sriov-network-operator` for vanilla k8s. And these CR manifests are more like examples. So I prefer to keep it as it is.

File: deploy/crds/sriovnetwork.openshift.io_v1_sriovibnetwork_cr.yaml [hunk 2]
```diff
@@ -0,0 +1,8 @@
...
+spec:
+  networkNamespace: default
+  resourceName: test
+  ipam: "{}"
```
[the remark this answers is not in the document]
Comment (pliurh) [reply deploy/crds/sriovnetwork.openshift.io_v1_sriovibnetwork_cr.yaml]: fixed.

**Comment by pliurh** [comment]
@Mmduh-483 As we haven't got the IB in our environment yet, may I ask if you could help to test this patch in your environment?

File: pkg/apis/sriovnetwork/v1/sriovibnetwork_types.go [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
+// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
+
+// SriovIBNetworkSpec defines the desired state of SriovIBNetwork
+type SriovIBNetworkSpec struct {
```
Comment (Mmduh-483) [inline pkg/apis/sriovnetwork/v1/sriovibnetwork_types.go:11]: There are other config fields for the ib-sriov, can you add them, i.e. `pkey`, `rdmaIsolation` and `ibKubernetesEnabled`

**Comment by Mmduh-483** [comment]
> @Mmduh-483 As we haven't got the IB in our environment yet, may I ask if you could help to test this patch in your environment?

I tested it this morning (without the last push changes) with the change linktype PR and generate random guid , it is working.

In the image that will be used for the ib-sriov please add this patch https://github.com/Mellanox/ib-sriov-cni/pull/41 (it will be merged soon) , it is needed so that the cni will skip the guid config if not passed in the cni config, the PR https://github.com/openshift/sriov-network-operator/pull/243 will set the guid for IB VFs.

**Comment by zshi-redhat** [comment]
/lgtm
@pliurh will you add webhook validation for sriov ib network?

**Comment by pliurh** [comment]
> /lgtm
> @pliurh will you add webhook validation for sriov ib network?

Yes.

**Pull request merged**